front/app/containers/Admin/projects/project/files/components/FilesUpload/components/SelectedFile.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+import {
+  Box,
+  Icon,
+  Text,
+  Spinner,
+  colors,
+  Select,
+  Tooltip,
+} from '@citizenlab/cl2-component-library';
+import { CLError } from 'typings';
+
+import useAddFile from 'api/files/useAddFile';
+
+import Error from 'components/UI/Error';
+
+import { useIntl } from 'utils/cl-intl';
+import { getBase64FromFile } from 'utils/fileUtils';
+
+import { FileWithMeta } from '../types';
+
+import messages from './messages';
+
+type Props = {
+  fileMeta: FileWithMeta;
+  projectId: string;
+  onStatusUpdate: (status: Partial<FileWithMeta>) => void;
+};
+
+const SelectedFile = ({ fileMeta, projectId, onStatusUpdate }: Props) => {
+  const { formatMessage } = useIntl();
+  const { mutateAsync } = useAddFile();
+  const { file, status, semanticType } = fileMeta;
+  const [apiErrors, setApiErrors] = useState<CLError[] | null>(null);
+
+  // Ref to track if the upload has started
+  const hasStarted = useRef(false);
+
+  // Function to upload the file
+  const uploadFile = useCallback(async () => {
+    const base64 = await getBase64FromFile(file);
+    await mutateAsync(
```

Comment:
Why do you use the mutateAsync function together with the onSuccess and orError handlers? I think we should stick to one or the other:
- mutateAsync in a try catch block
- mutate + onSuccess and onError handlers